### PR TITLE
Python Hashbangs

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Various discovered breakages 🙃.

### CODE REVIEW

File: Formula/gtk+.rb [hunk 1]
```diff
@@ -61,6 +62,8 @@ def install
     end
     system "./configure", *args
     system "make", "install"
+
+    inreplace bin/"gtk-builder-convert", "#!/usr/bin/env python", "#!/usr/bin/python"
```
Comment (ilovezfs) [inline Formula/gtk+.rb]: should use regex since we don't want to replace '#!/usr/bin/env python3' if it becomes that

Comment (DomT4) [reply Formula/gtk+.rb]: Ah, fair point. Do you want them all changed?

Comment (ilovezfs) [reply Formula/gtk+.rb]: yes please. I'd use `%r{^#!/usr/bin/env python$}`

Comment (DomT4) [reply Formula/gtk+.rb]: Will do. I see the bot has flagged more issues too so I'll take a peek at those whilst I'm here.

**Comment by DomT4** [comment]
The only remaining CI failure is the expected one:

```
lorem:
  * stable: sha256 changed without the version also changing; please create an issue upstream to rule out malicious circumstances and to find out why the file changed.
```

File: Formula/pygtkglext.rb [hunk 2]
```diff
@@ -28,7 +28,8 @@ def install
   end
 
   test do
-    ENV.append_path "PYTHONPATH", Formula["pygtk"].opt_lib+"python2.7/site-packages/gtk-2.0"
-    system "python", "-c", "import pygtk", "pygtk.require('2.0')", "import gtk.gtkgl"
+    ENV.append_path "PYTHONPATH", Formula["pygtk"].opt_lib/"python2.7/site-packages/gtk-2.0"
+    system Formula["python@2"].opt_bin/"python2.7", "-c", "import pygtk",
```
Comment (ilovezfs) [inline Formula/pygtkglext.rb]: why not use system Python?

File: Formula/pygtksourceview.rb [hunk 3]
```diff
@@ -27,7 +27,7 @@ def install
 
   test do
     ENV.append_path "PYTHONPATH", lib+"python2.7/site-packages"
-    ENV.append_path "PYTHONPATH", Formula["pygtk"].opt_lib+"python2.7/site-packages/gtk-2.0"
-    system "python", "-c", "import gtksourceview2"
+    ENV.append_path "PYTHONPATH", Formula["pygtk"].opt_lib/"python2.7/site-packages/gtk-2.0"
+    system Formula["python@2"].opt_bin/"python2.7", "-c", "import gtksourceview2"
```
Comment (ilovezfs) [inline Formula/pygtksourceview.rb]: ditto

File: Formula/pygtkglext.rb (hunk 2 above)
Comment (DomT4) [reply Formula/pygtkglext.rb]: Our `python@2` is part of the recursive deps tree via `pygobject` and `pygobject` uses `system Formula["python@2"].opt_bin/"python2.7"`, so essentially I went for consistency.

File: Formula/pygtksourceview.rb (hunk 3 above)
Comment (DomT4) [reply Formula/pygtksourceview.rb]: Same reply 😉

File: Formula/pygtkglext.rb (hunk 2 above)
Comment (ilovezfs) [reply Formula/pygtkglext.rb]: That is probably a bogus use of Homebrew Python so I don't think we need to further lock it in.

Comment (ilovezfs) [reply Formula/pygtkglext.rb]: I'd probably just make it `python2.7`

**Comment by ilovezfs** [comment]
Thanks!

**Pull request closed without merging**